src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { getTodos } from './api';
+import { actions as todosActions } from './features/todos';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const todos = useAppSelector(state => state.todos);
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+  const { query, status } = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+  const [isLoading, setLoading] = useState(false);
+
+  async function getTodosFromServer() {
+    setLoading(true);
+
+    try {
+      const data = await getTodos();
+
+      dispatch(todosActions.load(data));
+    } catch (error) {
+      window.console.log('Error: ', error);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  const visibleTodos = useMemo(() => {
+    if (status === 'all' && query !== '') {
```

Comment:
The condition in the if statement might not work as expected. It appears that you want to filter todos when the status is 'all' and there is a query. If that's the case, consider changing '&&' to '||' because the current condition will only return true if both conditions are met.

Reply:
@mykhalenych  this condition is corrected